tests/unit/test_runtime.py

```diff
@@ -48,30 +49,47 @@ def temp_dir(tmp_path_factory: TempPathFactory) -> str:
 PY3_FOR_TESTING = '/opendevin/miniforge3/bin/mamba run -n base python3'
 
 
-# This assures that all tests run together for each runtime, not alternating between them,
-# which caused them to fail previously.
-@pytest.fixture(scope='module')
-def box_class(request):
-    time.sleep(1)
+# Depending on TEST_RUNTIME, feed the appropriate box class(es) to the test.
+def get_box_classes():
     runtime = TEST_RUNTIME
     if runtime.lower() == 'eventstream':
-        return EventStreamRuntime
+        return [EventStreamRuntime]
     elif runtime.lower() == 'server':
-        return ServerRuntime
+        return [ServerRuntime]
     else:
-        return pytest.param([EventStreamRuntime, ServerRuntime])
+        return [EventStreamRuntime, ServerRuntime]
+
+
+# This assures that all tests run together per runtime, not alternating between them,
+# which cause errors (especially outside GitHub actions).
+@pytest.fixture(scope='module', params=get_box_classes())
+def box_class(request):
+    time.sleep(2)
+    return request.param
 
 
 # TODO: We will change this to `run_as_user` when `ServerRuntime` is deprecated.
 # since `EventStreamRuntime` supports running as an arbitrary user.
+# Prevent duplicate executions of tests that don't require run_as_devin.
+@pytest.fixture(scope='module')
+def run_as_devin():
+    return False  # default value for tests that don't care
+
+
 @pytest.fixture(scope='module', params=[True, False])
-def run_as_devin(request):
+def parameterized_run_as_devin(request):
```

Comment:
I'm not quite sure if this will work?

I didn't see any `run_as_devin` test being running in the CI though:
https://github.com/OpenDevin/OpenDevin/actions/runs/10178060624/job/28151939062?pr=3186

Reply:
yeh, that was a brainfart of Sonnet 😂 
I can't even run the full script locally as my WSL crashes in spite of 64 GB total RAM 😢